src/wp-includes/option.php

```diff
@@ -3214,3 +3214,126 @@ function wp_autoload_values_to_autoload() {
+ * It ensures that the metadata API handles this case correctly.
+ * This function is deprecated and advises using `add_network_option()` instead.
+ * If WordPress is not running in a multisite environment or if the `$unique` parameter is false,
+ * it will return `$current` or false as appropriate.
+ *
+ * @param mixed $current The return value of the original filter function.
+ * @param int $object_id ID of the object (site ID).
+ * @param string $meta_key Metadata key.
+ * @param mixed $meta_value Metadata value. Must be serializable if non-scalar.
+ * @param bool $unique Whether the metadata key should be unique. If false, `$current` is returned.
+ *
+ * @return bool|mixed False on failure, `$current` if the `$unique` parameter is false, or the result of `add_network_option()`.
+ */
+function filter_add_site_metadata( $current, $object_id, $meta_key, $meta_value, $unique ) {
+	if ( ! is_multisite() ) {
+		_doing_it_wrong( 'add_metadata', __( 'This function is only available in multisite installations.' ), '6.8.0' );
+		return false;
+	}
+
+	if ( ! $unique ) {
+		_doing_it_wrong( 'get_metadata', __( 'The $unique parameter is not supported in this function.' ), '6.8.0' );
+		return $current;
+	}
+
+	_doing_it_wrong( 'add_metadata', __( 'This function is deprecated. Use add_network_option() instead.' ), '6.8.0' );
+
+	return add_network_option( $object_id, $meta_key, $meta_value );
+}
+
+/**
+ * Filters site metadata deletion in multisite installations.
+ *
+ * This function ensures that metadata deletion adheres to multisite-specific rules.
+ * It also deprecates specific parameters and suggests using `delete_network_option()` instead.
+ *
+ * @param mixed $current The value to return if the function short-circuits.
+ * @param int $object_id The object ID, typically the site ID.
+ * @param string $meta_key The meta key to delete.
+ * @param mixed $meta_value The meta value to delete. Not supported in this function.
+ * @param bool $delete_all Whether to delete the matching metadata entries for all objects, ignoring the object ID. Not supported in this function.
+ *
+ * @return mixed The filtered result of the metadata deletion operation.
+ */
+function filter_delete_site_metadata( $current, $object_id, $meta_key, $meta_value, $delete_all ) {
+	if ( ! is_multisite() ) {
+		_doing_it_wrong( 'delete_metadata', __( 'This function is only available in multisite installations.' ), '6.8.0' );
+		return false;
+	}
+
+	if ( $delete_all ) {
+		_doing_it_wrong( 'delete_metadata', __( 'The $delete_all parameter is not supported in this function.' ), '6.8.0' );
+		return $current;
+	}
+
+	if ( $meta_value ) {
+		_doing_it_wrong( 'delete_metadata', __( 'The $meta_value parameter is not supported in this function.' ), '6.8.0' );
+		return $current;
+	}
+
+	_doing_it_wrong( 'delete_metadata', __( 'This function is deprecated. Use delete_network_option() instead.' ), '6.8.0' );
+
+	return delete_network_option( $object_id, $meta_key );
```

Comment:
Based on the above feedback, this can be simplified to:
```suggestion
	if ( $delete_all ) {
		return false;
	}

	if ( $meta_value ) {
		$option_value = get_network_option( $object_id, $meta_key, false );
		if ( $meta_value !== $option_value ) {
			return false;
		}
	}

	return delete_network_option( $object_id, $meta_key );
```